journal.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import os
+import datetime
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from zope.sqlalchemy import ZopeTransactionExtension
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from pyramid.security import remember, forget
+from cryptacular.bcrypt import BCRYPTPasswordManager
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+Base = declarative_base()
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(128), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    date = sa.Column(sa.DateTime, nullable=False,
+                        default=datetime.datetime.utcnow)
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.date.desc()).all()
+
+
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://johnson:password@localhost:5432/learning-journal'
+)
+
+
+def init_db():
+    engine = create_engine(DATABASE_URL)
+    Base.metadata.create_all(engine)
+
+
+@view_config(route_name='home', renderer='templates/lists.jinja2')
+def list_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='newpost', renderer='templates/newpost.jinja2')
+def new_post(request):
+    entries = Entry.all()
+    return {'entries': entries}
```

Comment:
Ahhhh, I see.  Why are you passing the full list of entries into a view that only displays a form for a new entry?  Doing too much work.